trikotFoundation/src/commonMain/kotlin/com/mirego/trikot/foundation/concurrent/dispatchQueue/SequentialDispatchQueue.kt

```diff
@@ -10,33 +10,24 @@ import com.mirego.trikot.foundation.concurrent.freeze
 open class SequentialDispatchQueue(override val dispatchQueue: TrikotDispatchQueue) :
     TrikotQueueDispatcher,
     TrikotDispatchQueue {
-    protected open val isSynchronous = false
     private val dispatchBlockQueue = AtomicListReference<DispatchBlock>()
     private val currentDispatch = AtomicReference(NoDispatchBlock)
 
     override fun isSerial() = true
 
     override fun dispatch(block: DispatchBlock) {
-        if (isSynchronous && currentDispatch.compareAndSet(NoDispatchBlock, SyncDispatchBlock)) {
-            dispatchQueue.dispatch(block)
-            currentDispatch.setOrThrow(SyncDispatchBlock, NoDispatchBlock)
-            startNextIfNeeded()
-        } else {
-            dispatchBlockQueue.add(block)
-            startNextIfNeeded()
-        }
+        dispatchBlockQueue.add(block)
+        startNextIfNeeded()
     }
 
     private fun startNextIfNeeded() {
-        if (currentDispatch.value == NoDispatchBlock) {
-            dispatchBlockQueue.value.firstOrNull()?.let { nextDispatchBlock ->
-                if (currentDispatch.compareAndSet(NoDispatchBlock, nextDispatchBlock)) {
-                    dispatchBlockQueue.remove(nextDispatchBlock)
-                    (this as TrikotQueueDispatcher).dispatch {
-                        nextDispatchBlock()
-                        markDispatchBlockCompleted(nextDispatchBlock)
-                        startNextIfNeeded()
-                    }
+        dispatchBlockQueue.value.firstOrNull()?.let { nextDispatchBlock ->
+            if (currentDispatch.compareAndSet(NoDispatchBlock, nextDispatchBlock)) {
+                dispatchBlockQueue.remove(nextDispatchBlock)
+                (this as TrikotQueueDispatcher).dispatch {
+                    nextDispatchBlock()
+                    markDispatchBlockCompleted(nextDispatchBlock)
+                    startNextIfNeeded()
```

Comment:
Since you already have the "dispatch lock", would it be better to loop until the dispatchBlockQueue is empty? Instead of calling `startNextIfNeeded` recursively?
You would save some CPU time releasing/acquiring the lock.